routers/api/v1/repo/tag.go

```diff
@@ -160,3 +162,62 @@ func DeleteTag(ctx *context.APIContext) {
 
 	ctx.Status(http.StatusNoContent)
 }
+
+// CreateTag create a new git tag in a repository
+func CreateTag(ctx *context.APIContext) {
+	// swagger:operation POST /repos/{owner}/{repo}/tags repository repoCreateTag
+	// ---
+	// summary: Create a new git tag in a repository
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the repo
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: path
+	//   description: name of the repo
+	//   type: string
+	//   required: true
+	// - name: body
+	//   in: body
+	//   schema:
+	//     "$ref": "#/definitions/CreateTagOption"
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/AnnotatedTag"
+	//   "404":
+	//     "$ref": "#/responses/notFound"
+	//   "409":
+	//     "$ref": "#/responses/conflict"
+	form := web.GetForm(ctx).(*api.CreateTagOption)
+
+	// If target is not provided use default branch
+	if len(form.Target) == 0 {
+		form.Target = ctx.Repo.Repository.DefaultBranch
+	}
+
+	commit, err := ctx.Repo.GitRepo.GetCommit(form.Target)
+	if err != nil {
+		ctx.Error(http.StatusNotFound, "target not found", fmt.Errorf("target not found: %v", err))
+		return
+	}
+
+	if err := releaseservice.CreateNewTag(ctx.User, ctx.Repo.Repository, commit.ID.String(), form.TagName, form.Message); err != nil {
```

Comment:
Just double check that this matches the commit api.

Often we allow people to set a title and a message body separately which would be:

form.Title + "\n\n" + form.Message

Reply:
@zeripath this func is already in use by webUI tag creation ... so it should work fine

Reply:
Yes and in the web we artificially split the title and the message:

https://github.com/go-gitea/gitea/blob/1bfb0a24d843e10d6d95c4319a84980485e584ed/routers/web/repo/release.go#L311

Reply:
However file API.FileOptions doesn't appear to separate things:

https://github.com/go-gitea/gitea/blob/ab77a24f18ae45248bd6c71778fe3bfacacf864f/modules/structs/repo_file.go#L9

So it's probably fine

Reply:
now I know what you ment ... yes this better as is for api ... make it consistent :)